chore(update): update modal added

WanderingHogan opened the pull request:
<!--  Thanks for sending a pull request!-->

### What this PR does 📖
Changes the 'update available' button to have a modal with directions so users don't get confused

![Screenshot 2023-04-28 at 3 47 54 PM](https://user-images.githubusercontent.com/2993032/235259796-ad45ee85-2020-41a2-a3ce-778100df4f4c.png)
![Screenshot 2023-04-28 at 3 48 25 PM](https://user-images.githubusercontent.com/2993032/235259863-2526879c-2516-4a77-bd43-3a3f653fac47.png)

- 

### Which issue(s) this PR fixes 🔨

- Resolve #

<!--Add the ticket Github number such as #Resolve #001 to automatically link the PR to the issue-->

### Special notes for reviewers 🗒️

### Additional comments 🎤

sdwoodbury committed:
port popup over from old uplink

sdwoodbury committed:
fix(clippy)

sdwoodbury committed:
saving progress

sdwoodbury committed:
wip: use Popup for software updates

sdwoodbury committed:
fix(updater): verify the correct installer has been released before prompting the user to download it

sdwoodbury committed:
fix linux installer

sdwoodbury committed:
rename to Modal and remove old css

sdwoodbury committed:
saving progress

WanderingHogan committed:
Update modal with instructions

phillsatellite committed:
Merge branch 'dev' into feat/add-popup

WanderingHogan committed:
fix cursor for phill

stavares843 commented on common/locales/en-US/main.ftl:
```diff
@@ -35,6 +35,14 @@ uplink = Uplink
+updates = Updates
+    .title = Uplink has an update available. Follow these steps to download and install the latest update.
+    .instruction1 = 1. Download the update.
+    .instruction2 = 2. Close uplink.
```
```suggestion
    .instruction2 = 2. Close Uplink.
```

stavares843 commented on common/locales/en-US/main.ftl:
```diff
@@ -35,6 +35,14 @@ uplink = Uplink
+    .instruction1 = 1. Download the update.
+    .instruction2 = 2. Close uplink.
+    .instruction3 = 3. Run the installer. The older version of uplink will be overwritten, leaving your account intact.
+    .instruction4 = 4. Open uplink and start chatting again!
```
```suggestion
    .instruction4 = 4. Open Uplink and start chatting again!
```

github-actions[bot] committed:
Merge dev into feat/add-popup

luisecm committed:
Merge branch 'dev' into feat/add-popup

InfamousVague committed:
Update common/locales/en-US/main.ftl

Co-authored-by: Sara Tavares <29093946+stavares843@users.noreply.github.com>

InfamousVague committed:
Update common/locales/en-US/main.ftl

Co-authored-by: Sara Tavares <29093946+stavares843@users.noreply.github.com>

InfamousVague committed:
Merge branch 'dev' into feat/add-popup

WanderingHogan committed:
updates to download modal

stavares843 commented on common/locales/en-US/main.ftl:
```diff
@@ -35,6 +35,15 @@ uplink = Uplink
+    .instruction4 = 4. Open Uplink and start chatting again!
+    .button-label = Pick Download Folder
+    .download-label = Download Update
+
```
do we have/are we gonna add force update?  🔨

WanderingHogan replied:
not sure
for now all manual

stavares843 commented on kit/src/components/modal/mod.rs:
```diff
@@ -0,0 +1,47 @@
+                        class: "model-body",
+                        rsx!(cx.props.children.as_ref()),
+                },
+
```
do we need these empty lines? 🔨

WanderingHogan replied:
no

stavares843 commented on ui/src/main.rs:
```diff
@@ -966,6 +974,70 @@ fn get_update_icon(cx: Scope) -> Element {
+            //            on_submit.call(dest.clone());
+            //         }
+            //     }
+            // ))
```
do we need this commented out code? 🔨

WanderingHogan replied:
we do, I didnt want to rip it out, it just wasn't working right now and I didnt have time to look into why

stavares843 commented:
how can we force the modal to appear? 🔨

sdwoodbury commented:
this could use a translation
![image](https://user-images.githubusercontent.com/8636602/235328032-30a49664-a8cb-446a-8f64-e5ac512d9fc6.png)

sdwoodbury committed:
fix(css): fix css class for download modal
the class was changed from flex to disp-flex to avoid conflicts, but the rust code didn't reflect that change

sdwoodbury committed:
chore(update): add translation for update modal

sdwoodbury committed:
fix(translation): remove quotes
